
OmniOS does not have an `ldap.pc` file for pkg-config to find so don't add it
to the private requirements list in `libcurl.pc`

Reverts upstream https://github.com/curl/curl/pull/15163

diff -wpruN --no-dereference '--exclude=*.orig' a~/CMakeLists.txt a/CMakeLists.txt
--- a~/CMakeLists.txt	1970-01-01 00:00:00
+++ a/CMakeLists.txt	1970-01-01 00:00:00
@@ -1042,7 +1042,6 @@ if(NOT CURL_DISABLE_LDAP)
       list(APPEND CMAKE_REQUIRED_DEFINITIONS "-DLDAP_DEPRECATED=1")
       list(APPEND CMAKE_REQUIRED_LIBRARIES ${LDAP_LIBRARY})
       set(CURL_LIBS "${LDAP_LIBRARY};${CURL_LIBS}")
-      set(LIBCURL_PC_REQUIRES_PRIVATE "ldap;${LIBCURL_PC_REQUIRES_PRIVATE}")
       if(HAVE_LIBLBER)
         list(APPEND CMAKE_REQUIRED_LIBRARIES ${LDAP_LBER_LIBRARY})
         set(CURL_LIBS "${LDAP_LBER_LIBRARY};${CURL_LIBS}")
diff -wpruN --no-dereference '--exclude=*.orig' a~/acinclude.m4 a/acinclude.m4
--- a~/acinclude.m4	1970-01-01 00:00:00
+++ a/acinclude.m4	1970-01-01 00:00:00
@@ -565,7 +565,6 @@ AC_DEFUN([CURL_CHECK_LIBS_LDAP], [
       else
         LIBS="$curl_cv_ldap_LIBS $curl_cv_save_LIBS"
       fi
-      LIBCURL_PC_REQUIRES_PRIVATE="ldap $LIBCURL_PC_REQUIRES_PRIVATE"
       AC_MSG_RESULT([$curl_cv_ldap_LIBS])
       ;;
   esac
